Library/Homebrew/formulary.rb

```diff
@@ -136,6 +136,17 @@ def self.load_formula_from_api(name, flags:)
     class_s = Formulary.class_s(name)
     json_formula = Homebrew::API::Formula.all_formulae[name]
 
+    bottle_tag = Utils::Bottles.tag.to_s
+    if json_formula.key?("variations") && json_formula["variations"].key?(bottle_tag)
+      json_formula = json_formula.merge(json_formula["variations"][bottle_tag])
+    end
```

Comment:
```suggestion
    
    if (bottle_tag = Utils::Bottles.tag.to_s.presence) && 
       (variations = json_formula["variations"].presence) &&   
       (variation = variations[bottle_tag].presence)
      json_formula = json_formula.merge(variation)
    end
```

avoids triggering any of this logic if there's a `blank?` value at any point and avoids rereading/checking the same data multiple times

Reply:
Sure, good idea. Is there ever an (intentional) situation where `Utils::Bottles.tag` might be blank?

Reply:
Nah, shoudn't be, that's probably me being unnecessarily cautious, was just thinking that the variable is only needed in this scope.

Reply:
👍 I think it makes sense, and that way formulae can still be loaded even if something crazy happens